R/covr.R

```diff
@@ -429,3 +455,23 @@ add_hooks <- function(pkg_name, lib, fix_mcexit = FALSE) {
 
   writeLines(text = lines, con = load_script)
 }
+
+# check if gcc or icc is used
+get_compiler <- function() {
```

Comment:
I think this function should just check if `icc` is used and fallback to the current behavior otherwise. Also you cannot assume the `R CMD config CC` will have trailing spaces, using the  [word boundary](http://www.regular-expressions.info/wordboundaries.html) `\\b` anchor is a better solution.

```r
uses_icc <- function() {
  compiler <- tryCatch(
    paste(system(paste(R.home("bin"), "R --vanilla CMD config CC", sep="/"),
        intern = TRUE), collapse=""),
    error = function(e) "")
  grepl("\\bicc\\b", compiler)
}
```